src/jdb_to_nwb/convert_video.py

```diff
@@ -1,11 +1,178 @@
 from pynwb import NWBFile
+from scipy.ndimage import gaussian_filter
+import csv
+import numpy as np
 
+def add_video(nwbfile: NWBFile, metadata: dict, photometry_start_in_arduino_time = 55520059.6736):
+    print("Adding video..")
 
+    # Get file paths for video from metadata file
+    video_file_path = metadata["video"]["arduino_video_file_path"]
+    video_timestamps_file_path = metadata["video"]["arduino_video_timestamps_file_path"]
 
+    # metadata should include ethe full name of dlc algorithm
+    phot_dlc = metadata["video"]["dlc_algorithm"] # Behav_Vid0DLC_resnet50_Triangle_Maze_EphysDec7shuffle1_800000.h5
 
-def convert_video(nwbfile: NWBFile, metadata: dict):
+    # Get pixelsPerCm based on the date of the data collected
+    pixelsPerCm = assign_pixels_per_cm(metadata["date"])
 
+    # Read times of each position point, equal to the time of each camera frame recording
+    with open(video_timestamps_file_path, "r") as video_timestamps_file:
+        video_timestamps = np.array(list(csv.reader(open(video_timestamps_file, 'r'))), dtype=float).ravel()
+
+    # Convert arduino timestamps to corresponding photosmetry sample number
+    video_timestamps = adjust_video_timestamps(video_timestamps, photometry_start_in_arduino_time)
+
+    # Read x and y position data and calculate velocity and acceleration
+    x, y, velocity, acceleration = read_dlc(deeplabcut_file_path, phot_dlc = phot_dlc, cutoff = 0.9, cam_fps = 15, pixelsPerCm)
+
+    return video_timestamps, x, y, velocity, acceleration
+
+def assign_pixels_per_cm(date_str):
+    """
+    Assigns pixelsPerCm based on the provided date string in mmddyyyy format.
+    - date_str (str): Date string in 'mmddyyyy' format, e.g., '11122022'.
+
+    3.14 if video is before IM-1594(before 01012023). 2.3 before (01112024). 2.688 after (old maze)
+
+    Returns:
+    - float: The corresponding pixelsPerCm value.
```

Comment:
make PIXELS_PER_CM uppercase in function documentation
```suggestion
    Assigns constant PIXELS_PER_CM based on the provided date string in MMDDYYYY format.
    PIXELS_PER_CM is 3.14 if video is before IM-1594 (before 01012023), 2.3 before (01112024), or 2.688 after (old maze)
    
    Args:
    - date_str (str): Date string in MMDDYYYY format, e.g., '11122022'.

    Returns:
    - float: The corresponding PIXELS_PER_CM value.
```